### PR TITLE
UI Confirm Dissolution Modal Fixes/11663

### PR DESCRIPTION
**Pull request by meganawong**
*Issue #:* /bcgov/entity#11663

*Description of changes:*
* fix confirm dissolution modal for SP/GP 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/resources/business-config.ts [hunk 1]
```diff
@@ -80,7 +80,8 @@ export const ConfigJson = [
       act: 'Business Corporations',
       modalTitle: 'Voluntary Dissolution',
       confirmButtonText: 'Continue with Voluntary Dissolution',
-      dissolutionType: 'voluntarily dissolve'
+      dissolutionType: 'voluntarily dissolve',
+      incorporationText: 'an incorporated'
```
Comment (severinbeauvais) [inline src/resources/business-config.ts:84]: I _think_ `incorporationText` and `subTitle` are only used in one place, and always together. Why not combine them?

File: src/components/dialogs/ConfirmDissolution.vue [hunk 2]
```diff
@@ -100,6 +100,11 @@ export default class ConfirmDissolution extends Vue {
     return this.getDissolutionConfirmationResource?.dissolutionType
   }
 
+  /** The incorporation text to display. */
+  private get incorporationText (): string {
```
Comment (severinbeauvais) [inline src/components/dialogs/ConfirmDissolution.vue:104]: Small cleanup: please change the local getters in this file to public (just remove private).

Also please change the prop to readonly.

**Review by severinbeauvais — COMMENTED** [review]
Unit tests should have broken this change... but I see there are no unit tests for the various warning text scenarios.

Please add one (you only need to pass in props and verify that the warning text uses them correctly).

File: src/resources/business-config.ts (hunk 1 above)
Comment (meganawong) [reply src/resources/business-config.ts]: combined

File: src/components/dialogs/ConfirmDissolution.vue (hunk 2 above)
Comment (meganawong) [reply src/components/dialogs/ConfirmDissolution.vue]: done

File: src/components/dialogs/ConfirmDissolution.vue [hunk 3]
```diff
@@ -58,45 +58,45 @@ import { Getter } from 'vuex-class'
...
-  private get entityTitle (): string {
+  get entityTitle (): string {
     return this.getDissolutionConfirmationResource?.entityTitle
   }
 
   /** The entity title to display. */
-  private get subEntityTitle (): string {
-    return this.getDissolutionConfirmationResource?.subTitle
+  get subEntityTitle (): string {
+    return this.getDissolutionConfirmationResource.subTitle
```
Comment (severinbeauvais) [inline src/components/dialogs/ConfirmDissolution.vue:75]: You removed the `?`, so if getDissolutionConfirmationResource is undefined/null then this getter will fail.

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

**Review by severinbeauvais — APPROVED** [review]
OK to merge after optional chaining operator is restored.

**Pull request merged**